app/test_main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+import pytest
+
+from unittest import mock
+
+from app.main import can_access_google_page
+
+from typing import Callable
+
+
+@pytest.mark.parametrize(
+    "is_connected,is_valid_url,expected",
+    [
+        (True, True, "Accessible"),
+        (False, True, "Not accessible"),
+        (True, False, "Not accessible"),
+        (False, False, "Not accessible")
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
```

Comment:
Change the decorators' sequence. 
It will be more logical if you first check whether the user has an internet connection